daemon/lyrics.py

```diff
@@ -100,7 +108,8 @@ def decode_by_charset(content):
     >>> decode_by_charset(u'\u4e2d\u6587'.encode('HZ-GB-2312'))
     u'\u4e2d\u6587'
     """
-    if not isinstance(content, str):
+    # TODO: bail out if the type IS unicode (Python2) / str (Python3)
+    if not isinstance(content, NOT_UNICODE_TYPE):
```

Comment:
I left the TODO above so we can focus on this during review. I would like to have this simplified:
`if isinstance(content, UNICODE_TYPE):` where `UNICODE_TYPE` would be `str` under Python 3 and `unicode` under Python 2. I'll test such a change and report back.